app/services/stepfunctions/RegularContributionsClient.scala

```diff
@@ -169,7 +169,7 @@ object StepFunctionExecutionStatus {
     val searchForFinishedCheckout: Option[StatusResponse] = detailedHistory.collectFirst {
       case detailsAttempt if detailsAttempt.map(_.getName) == Success("CheckoutSuccess") =>
         StatusResponse(Status.Success, trackingUri, None)
-      case detailsAttempt if detailsAttempt.map(_.getName) == Success("CheckoutFailure") =>
```

Comment:
We are still going to need to handle cases where we reach the `CheckoutFailure` state as this is the more common one caused by card processing type errors

Reply:
we do, this state history is a list and this state comes before both so it will behave the same as before for the old state

Reply:
adding to what @johnduffell said, the state history does not contain the 'FailState` even if it does reach it. This is presumably because it has `Type: Fail`
So `SucceedOrFailChoice` allows us to catch both

Reply:
👍

Reply:
They might well still end up seeing a thank you page if the execution takes more than the timeout period, but this is definitely and improvement